### PR TITLE
Add Riffraff integration doc

### PR DESCRIPTION
**Pull request by thomaska**
Added some (hopefully helpful!) documentation on how to use Amigo baked instances with RiffRaff.

This is a first attempt to add some more documentation on how to use Amigo baked AMIs in Riffraff. I hope it makes some sense! Would appreciate any feedback! @sihil @adamnfish @TBonnin !

### CODE REVIEW

**Comment by thomaska** [comment]
[Add more documentation on Amigo](https://trello.com/c/ED9cjERp/20-add-more-documentation-on-amigo)

File: docs/riffraff-integration.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+	            "cloudFormationStackName": "cloudformation-stack-name",
+	            "amiParametersToTags": {
+	              "AMI" : {
+	                "Recipe": "ubuntu-wily-java8",
+	                "AmigoStage":"PROD"
+	              }
+	            }
+	          },
+	          ...
+        }
```
Comment (sihil) [inline docs/riffraff-integration.md]: People shouldn't be using the JSON format anymore, so I'd prefer it if we simply didn't document it.

File: docs/riffraff-integration.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+After doing that we would like Riffraff to check in every deployment whether there is a a newer AMI than the one currently used. In order to do that, RiffRaff should know for which image we are talking about (eg. Ubuntu Xenial with Java 8) and for which cloudformation script.
+ -	In order to define the AMI: We must define its tags in our `deploy.json` or `riff-raff.yaml` file. This can be done either using the `amiTags` parameter or the `amiParametersToTags` parameter, if we use multiple AMIs in one cloudformation script or if we want to define explicitly the name of the AMI parameter. 
+ -	In order to define the cloudformation script we can either define it using tags or its name. When using tags we find the stack 	by looking for one with matching stack, app and stage tags in the same way that autoscaling groups are discovered. When using the cloudformation name we set the parameter `cloudFormationStackName`. To set which of the two ways to use, we must set the parameter `cloudFormationStackByTags`.
+
+	For more detailed description of the parameters please check here: 
+	https://riffraff.gutools.co.uk/docs/magenta-lib/types#amicloudformationparameter
+
+	Examples:
+
+		```
```
Comment (sihil) [inline docs/riffraff-integration.md]: Seems that these triple backticks are visible in the rendered markdown. Guess you only need one of indentation _or_ backticks?

File: docs/riffraff-integration.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+		  update-ami:
+		    type: ami-cloudformation-parameter
+		    parameters:
+		      cloudFormationStackByTags: false
+		      prependStackToCloudFormationStackName: false
+		      cloudFormationStackName: cloudformation-stack-name
+		      amiParametersToTags:
+		        AMI:
+		          Recipe: ubuntu-wily-java8
+		          AmigoStage: PROD
```
Comment (sihil) [inline docs/riffraff-integration.md]: As previously discussed with @TBonnin my opinion is that we should encourage people to be specific where possible. In this case I'd prefer it if people included a `BuiltBy: Amigo` tag in the list.

File: docs/riffraff-integration.md (hunk 2 above)
Comment (adamnfish) [inline docs/riffraff-integration.md]: #protip if you use the block syntax (three backticks) you can set the highlighting scheme using a language identifier, by following the opening backticks with the name (i.e. `` ```scala``, `` ```yaml`` [etc](https://github.com/github/linguist/blob/master/lib/linguist/languages.yml))

File: docs/riffraff-integration.md (hunk 1 above)
Comment (philwills) [inline docs/riffraff-integration.md]: 👍 for focusing on the new format

**Pull request merged**